Doc/library/importlib.metadata.rst

```diff
@@ -406,6 +406,84 @@ metadata in locations other than the file system, subclass
 a custom finder, return instances of this derived ``Distribution`` in the
 ``find_distributions()`` method.
 
+Example
+-------
+
+Consider for example a custom finder that loads Python
+modules from a database::
+
+    class DatabaseImporter(importlib.abc.MetaPathFinder):
+        def __init__(self, db):
+            self.db = db
+
+        def find_spec(self, fullname, target=None) -> ModuleSpec:
+            return self.db.spec_from_name(fullname)
+
+    sys.meta_path.append(DatabaseImporter(connect_db(...)))
+
+That importer now presumably provides importable modules from a
+database, but it provides no metadata or entry points. For this
+custom importer to provide metadata, it would also need to implement
+``DistributionFinder``::
```

Comment:
This says that to provide metadata, one must *also* implement `DistributionFinder` but the parent class, `importlib.abc.MetaPathFinder`, is replaced by `DistributionFinder` in the snippet below.  The reader must infer that `DistributionFinder` is a subclass of `importlib.abc.MetaPathFinder`.

Reply:
I can see a couple of ways to highlight this detail. One could change the narrative to something like:

> ...it would also need to implement `DistributionFinder` (a subclass of MetaPathFinder):

Alternatively, the `DistributionFinder` could be changed from a raw identifier to a reference to `:class:DistributionFinder`, which [captures that detail](https://importlib-metadata.readthedocs.io/en/latest/api.html#importlib_metadata.DistributionFinder). That, of course, would require that the API docs be implemented, which is something I'm reluctant to do because it's something [a machine can do](https://github.com/python/importlib_metadata/blob/main/docs/api.rst) (just not in the context of CPython AFAIK). I guess the CPython docs could just link back to the importlib_metadata docs, i.e.:

```
...it would also need to implement `DistributionFinder <https://importlib-metadata.readthedocs.io/en/latest/api.html#importlib_metadata.DistributionFinder>`_.
```

WDYT?

Reply:
Personally, I would much rather the full documentation gets consolidated into the CPython documentation, rather than calling out to the `importlib_metadata` docs. While I appreciate this is extra work, I think it is worthwhile nevertheless, because it brings the `importlib.metadata` in line with the rest of the CPython documentation (in terms of style, maintenance workflow, and ease of access).

It's not me that would be doing the work, though, so I'm just offering this as my viewpoint. If you feel as maintainer that the cost isn't justified by the benefits, then that's your call.